src/node/internal/internal_timers.ts

```diff
@@ -0,0 +1,320 @@
+import { validateFunction, validateNumber } from 'node-internal:validators';
+import { ERR_OUT_OF_RANGE } from 'node-internal:internal_errors';
+
+// Timeout values > TIMEOUT_MAX are set to 1.
+const TIMEOUT_MAX = 2 ** 31 - 1;
+
+let clearTimeoutImpl: (obj: Timeout) => void;
+let clearImmediateImpl: (obj: Immediate) => void;
+
+function getTimerDuration(msecs: unknown, name: string): number {
+  validateNumber(msecs, name);
+  if (msecs < 0 || !Number.isFinite(msecs)) {
+    throw new ERR_OUT_OF_RANGE(name, 'a non-negative finite number', msecs);
+  }
+
+  // Ensure that msecs fits into signed int32
+  if (msecs > TIMEOUT_MAX) {
+    return TIMEOUT_MAX;
```

Comment:
Does't do what the comment says: `// Timeout values > TIMEOUT_MAX are set to 1.`